### PR TITLE
feat(client): Start on the FxFennecV1AuthenticationBroker

### PR DESCRIPTION
**Pull request by shane-tomlinson**
issue #2771 

This is going to be for @ncalexan and the work he's doing with Fennec.

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@zaach - tests added, ready for review!

**Comment by ncalexan** [comment]
@zaach, @stomlinson -- thanks for knocking this out so quickly.  Y'all rock.

**Comment by zaach** [comment]
Reports show a Mocha test error:
```
Failed test: "before each" hook
 => User is not a constructor
```

File: app/tests/spec/models/auth_brokers/fx-fennec-v1.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  'chai',
+  'lib/channels/null',
+  'models/auth_brokers/fx-fennec-v1',
+  'models/reliers/relier',
+  'models/user',
+  'sinon',
+  '../../../mocks/window'
+],
+function (chai, NullChannel, FxFennecV1AuthenticationBroker, Relier,
+  sinon, User, WindowMock) {
```
Comment (zaach) [inline app/tests/spec/models/auth_brokers/fx-fennec-v1.js]: `sinon` and `User` need to be swapped to match the order of the paths.

**Comment by shane-tomlinson** [comment]
> sinon and User need to be swapped to match the order of the paths.

Doh. Updated and re-pushed.

**Comment by zaach** [comment]
:+1: :100:

**Pull request merged**